New package: openrgb-0.5

NeelChotai opened the pull request.

DirectorX commented:
Is this software have daemon mode?

DirectorX commented:
You may want to set the `archs=` variable, those RGB controllers are tided to x86 systems anyway.

NeelChotai commented:
> 
> 
> Is this software have daemon mode?

I believe this will be a feature in the next update, thus far I've been using the CLI functionality in my window manager launch script.

> 
> 
> You may want to set the `archs=` variable, those RGB controllers are tided to x86 systems anyway.

I considered this but the program has support for USB RGB devices too. I'm not sure about inherent incompatibilities with other architectures.

Something seems to be wrong with Travis - it's manually compiling dependencies. I'll have another crack at this soon.

pullmoll commented:
@DirectorX there is no need to restrict with `archs=...` just because (currently) a package is useful only for specific architectures. There may be (1) use cases which do not require a device, or (2) cross platform use like with e.g. `mac-fdisk` which could be used to edit an Apple partition table on a device connected to an x86_64 system.

The general rule for using `archs=...` is this: Specify `__`**only**`__` in case a package contains e.g. assembler code for just specific target architectures and as a result cannot be built for other platforms. If it can be built, do not restrict archs.

Eluminae commented:
maybe we should setup the udev module as https://gitlab.com/CalcProgrammer1/OpenRGB#usb-access-1

NeelChotai commented:
> 
> 
> maybe we should setup the udev module as https://gitlab.com/CalcProgrammer1/OpenRGB#usb-access-1

I omitted this since it wasn't released in 0.2 but I will add it in manually.

I'll do so when I have a go at fixing cross tonight.

NeelChotai commented:
Compiles successfully and confirmed working on x86_64.

Version 0.2 used system groups to control RGB which has been superseded by the udev rules in master. If anyone else can test and ensure that OpenRGB works without root permissions, this is good to merge.

NeelChotai commented:
Updated to 0.3, this should be ready to merge.

Piraty reviewed (changes requested):
1. `0.4` is out
1. please name it all lowercase (see [repology](https://repology.org/projects/?search=OpenRGB))
1. take some inspiration from #24150.

NeelChotai commented:
>     1. `0.4` is out
> 
>     2. please name it all lowercase (see [repology](https://repology.org/projects/?search=OpenRGB))
> 
>     3. take some inspiration from #24150.

Updated to 0.4, thanks for the reminder.

I'm going by the Manual, the tar.gz is uppercase and packages in some other repos are uppercase too. If you believe renaming it lowercase is still preferable, let me know and I'll change it.

Added the service and log files from #24150, thank you @cpixl.

Piraty commented:
please lowercase it, else it's good to merge

NeelChotai commented:
@Piraty the OpenRGB developers have made the decision to lowercase the binary so I've taken the liberty of lowercasing everything for this pull request, it should be good to merge now.

Piraty commented on srcpkgs/openrgb/template:
```diff
@@ -0,0 +1,24 @@
+checksum=e227dedfe0c3aa8f3bcb0c4149aa5feb1db4b0429a151423d74c0103c55d7d26
+
+pre_build() {
+	sed -i 's|rules.path=/lib|rules.path=/usr/lib|g' OpenRGB.pro
```
vsed please

Piraty reviewed (changes requested):
else it's good to go
